typescript/sdk/src/deploy/verify/ZKSyncContractVerifier.ts

```diff
@@ -0,0 +1,158 @@
+import { ChainName } from '../../types.js';
+
+import { BaseContractVerifier } from './BaseContractVerifier.js';
+import {
+  BuildArtifact,
+  ContractVerificationInput,
+  SolidityStandardJsonInput,
+  ZKSyncCompilerOptions,
+} from './types.js';
+
+/**
+ * @title ZKSyncContractVerifier
+ * @notice Handles the verification of ZkSync contracts on block explorers
+ * @dev This class manages the process of verifying ZkSync contracts, including
+ * preparing verification data and submitting it to the appropriate explorer API
+ * Note: Etherscan verification is managed by the ContractVerifier class
+ * Blockscout verification is not currently supported on ZkSync
+ */
+export class ZKSyncContractVerifier extends BaseContractVerifier {
+  protected logger = rootLogger.child({ module: 'ZKSyncContractVerifier' });
+
+  protected readonly standardInputJson: SolidityStandardJsonInput;
+  protected readonly compilerOptions: ZKSyncCompilerOptions;
+
+  /**
+   * @notice Creates a new ZKSyncContractVerifier instance
+   * @param multiProvider An instance of MultiProvider for interacting with multiple chains
+   */
+  constructor(protected readonly multiProvider: MultiProvider) {
+    super(multiProvider, buildArtifact);
+    this.standardInputJson = (buildArtifact as BuildArtifact).input;
+
+    const compilerZksolcVersion = `v${
+      (buildArtifact as { zk_version: string }).zk_version
+    }`;
+    const compilerSolcVersion = (buildArtifact as BuildArtifact)
+      .solcLongVersion;
+
+    this.compilerOptions = {
+      codeFormat: 'solidity-standard-json-input',
+      compilerSolcVersion,
+      compilerZksolcVersion,
+      optimizationUsed: true,
+    };
+  }
+
+  /**
+   * @notice Verifies a contract on the specified chain
+   * @param chain The name of the chain where the contract is deployed
+   * @param input The contract verification input data
+   * @param verificationLogger A logger instance for verification-specific logging
+   */
+  protected async verify(
+    chain: ChainName,
+    input: ContractVerificationInput,
+    verificationLogger: Logger,
+  ): Promise<void> {
+    const contractType: string = input.isProxy ? 'proxy' : 'implementation';
+
+    verificationLogger.debug(`📝 Verifying ${contractType}...`);
+
+    const data = this.getImplementationData(chain, input, verificationLogger);
+
+    try {
+      const verificationId: string = await this.submitForm(
+        chain,
+        verificationLogger,
+        data,
+      );
+
+      verificationLogger.trace(
+        { verificationId },
+        `Retrieved verificationId from verified ${contractType}.`,
+      );
+    } catch (error) {
+      verificationLogger.debug(
+        { error },
+        `Verification of ${contractType} failed`,
+      );
+      throw error;
+    }
+  }
+
+  protected prepareImplementationData(
+    sourceName: string,
+    input: ContractVerificationInput,
+    filteredStandardInputJson: SolidityStandardJsonInput,
+  ) {
```

Comment:
This should be typed to return `FormOptions<VERIFY_IMPLEMENTATION>` but apparently, `SolidityStandardJsonInput` is not a string, so not assignable to `sourceCode`. Do we want to make our types more sophisticated here?

`ContractVerifier` does a `JSON.stringify()` for this field. Is ZKSync different in this way?